### ISSUE TITLE
linter: `typescript/consistent-type-imports` rule is not working

### ISSUE THREAD

**Issue by Spielboerg** [oxc-project/oxc#23971]
### What version of Oxlint are you using?

1.71.0

### What command did you run?

`oxlint -c ./oxlint.config.ts`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```ts
import { defineConfig } from 'oxlint'

export default defineConfig({
  rules: {
    'no-console': 'error', // Just to see that Oxlint actually runs
    'typescript/consistent-type-imports': 'error',  // Default `fixStyle`: 'separate-type-imports'
  },
})
```

### What happened?

The rule [`typescript/consistent-type-imports`](https://oxc.rs/docs/guide/usage/linter/rules/typescript/consistent-type-imports.html) doesn't work at all.

Reproduction:

```tsx
// Test.tsx

import { type ComponentProps } from 'react'

const Test = (props: ComponentProps<'div'>) => {
  console.log('Test component')

  return <div {...props}>Test</div>
}

export { Test }
```

Run `oxlint ./Test.tsx -c ./oxlint.config.ts`

Linter output:
```
  × eslint(no-console): Unexpected console statement.
   ╭─[file:///<redacted path>/Test.tsx:4:3]
 3 │ const Test = (props: ComponentProps<'div'>) => {
 4 │   console.log('Test component')
   ·   ───────────
 5 │ 
   ╰────
  help: Delete this console statement.

Found 0 warnings and 1 error.
Finished in 32ms on 1 file with 97 rules using 10 threads.
```

Additional expected output (or auto-fixed when run with `--fix`):
_(I don't know what the error would look like, so this is a made-up error message)_
```
import { type ComponentProps } from 'react'
         ^^^^

Add the `type` keyword after `import`:

import type { ComponentProps } from 'react'
```

**Comment by camc314** [comment]
hi! Thanks for opening this issue

`consistent-type-imports` doesn't enforce preferring `import type {...` over `import { type§.

I would suggest using `consistent-type-specifier-style` (which mirrors the `import/consistent-type-specifier-style` rule)
https://oxc.rs/docs/guide/usage/linter/rules/import/consistent-type-specifier-style.html 


Please let me know if this is not the case - Thanks

**Comment by Spielboerg** [comment]
@camc314 thank you! The suggested rule works for me and does what I want to achieve.

I think the docs for `consistent-type-imports` could be improved, because if I understand it correctly, it just adds missing `type` keywords, right? And the `fixStyle` setting lets you specify where they should be placed initially? If I'm right, that's not entirely clear, which is why I'm suggesting the following for the documentation page:

```diff
### What it does

- Enforce consistent usage of type imports.
+ Enforce consistent usage of type imports by adding or removing the `type` keyword from imports.
+ Note that this rule does not change the position of existing `type` keywords.
+ To enforce consistent placement, you can use [`import/consistent-type-specifier-style`](https://oxc.rs/docs/guide/usage/linter/rules/import/consistent-type-specifier-style.html).
```

**Comment by camc314** [comment]
Thanks for the suggestion - will update the docs!
